lucene/core/src/test/org/apache/lucene/search/TestSort.java

```diff
@@ -878,4 +878,22 @@ public void testMultiSort() throws IOException {
     ir.close();
     dir.close();
   }
+
+  public void testRewrite() throws IOException {
+    try (Directory dir = newDirectory()) {
+      RandomIndexWriter writer = new RandomIndexWriter(random(), dir);
+      IndexSearcher searcher = newSearcher(writer.getReader());
+      writer.close();
+
+      LongValuesSource longSource = LongValuesSource.constant(1L);
+      Sort sort = new Sort(longSource.getSortField(false));
+
+      assertSame(sort, sort.rewrite(searcher));
+
+      DoubleValuesSource doubleSource = DoubleValuesSource.constant(1.0);
+      sort = new Sort(doubleSource.getSortField(false));
+
+      assertSame(sort, sort.rewrite(searcher));
+    }
+  }
```

Comment:
This test looks like it belongs to TestDoubleValuesSource/TestLongValuesSource more than it belongs to TestSort?